synthesis_alu/testbench.cpp

```diff
@@ -0,0 +1,246 @@
+/******************************************************************************
+ * Copyright (c) 2020, Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception.
+ *
+ *****************************************************************************/
+
+// Design template
+
+#include "alu.cpp"
+#include <systemc.h>
+
+// ICSC requires DUT top should be instantiated inside wrapper (typically TB)
+// and all DUT ports are bound.
+struct Tb : sc_module {
+    sc_in<bool> clk;
+    sc_signal<bool> reset;
+    sc_signal<sc_bv<32>> rs1;
+    sc_signal<sc_bv<32>> rs2;
+    sc_signal<sc_bv<12>> imm;
+    sc_signal<sc_bv<17>> instruction;
+
+    sc_signal<sc_bv<32>> rd;
+
+    alu dut_inst{"dut_inst"};
+
+    SC_CTOR(Tb) {
+        dut_inst.clk(clk);
+        dut_inst.reset(reset);
+        dut_inst.rs1(rs1);
+        dut_inst.rs2(rs2);
+        dut_inst.imm(imm);
+        dut_inst.instruction(instruction);
+        dut_inst.rd(rd);
+
+        SC_THREAD(test);
+    }
+
+    void test() {
+        const int OP_ADD = 0b00000000000110011;
```

Comment:
Wäre hier unsigned oder so nicht besser? Oder was sagt das Synthesizable Subset?

Reply:
Ist beides erlaubt. Da ich in der Alu wieder nur die Binärdarstellung zum Vergleichen nehme ist egal, wie er den int dahinter interpretiert